release-22.2: changefeedccl: Improve error handling

miretskiy opened the pull request:
Backport 1/2 commits from #106421.

Backport required some conflict resolution.  

Release justification: Improve error reporting for customer issue.

/cc @cockroachdb/release

changefeedccl: Treat kv decoding errors as terminal 

Treat key/value decoding errors as terminal.
Produce loud errors when encountering such an error.

Fixes #106384

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114152)
<!-- Reviewable:end -->

miretskiy commented:
Informs https://github.com/cockroachdb/cockroach/issues/114133

miretskiy commented:
@nicktrav -- NOOO! these are not expected.  Hate unclean backports.  Thanks.

Commit:
changefeedccl: Treat kv decoding errors as terminal

Treat key/value decoding errors as terminal.
Produce loud errors when encountering such an error.

Fixes #106384

Release note: None

nicktrav commented:
I did a re-read. Still EM LGTM ✅.
